build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import com.adarshr.gradle.testlogger.TestLoggerExtension
+import com.adarshr.gradle.testlogger.TestLoggerPlugin
+import org.jetbrains.intellij.IntelliJPluginExtension
+import org.jetbrains.intellij.tasks.PrepareSandboxTask
+import org.jetbrains.intellij.tasks.PublishTask
+import org.jetbrains.intellij.tasks.RunIdeTask
+import org.jetbrains.intellij.tasks.RunIdeForUiTestTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import software.aws.toolkits.gradle.IdeVersions
+import software.aws.toolkits.gradle.ProductCode
+import software.aws.toolkits.gradle.changelog.tasks.GenerateGithubChangeLog
+import software.aws.toolkits.gradle.findFolders
+import software.aws.toolkits.gradle.getOrCreate
+import software.aws.toolkits.gradle.resources.ValidateMessages
+
+buildscript {
+    repositories {
+        maven("https://plugins.gradle.org/m2/")
+        mavenCentral()
+        jcenter()
+    }
+    val kotlinVersion: String by project
+    val ideaPluginVersion: String by project
+    dependencies {
+        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
+        classpath("gradle.plugin.org.jetbrains.intellij.plugins:gradle-intellij-plugin:$ideaPluginVersion")
+        classpath("com.adarshr:gradle-test-logger-plugin:1.7.0")
+    }
+}
+
+val ideVersions = IdeVersions(project)
+val ideVersion = ideVersions.resolveShortenedIdeProfileName()
+val toolkitVersion: String by project
+val kotlinVersion: String by project
+val mockitoVersion: String by project
+val mockitoKotlinVersion: String by project
+val assertjVersion: String by project
+val junitVersion: String by project
+val remoteRobotPort: String by project
+val ktlintVersion: String by project
+val remoteRobotVersion: String by project
+val ideaPluginVersion: String by project
+// publishing
+val publishToken: String by project
+val publishChannel: String by project
```

Comment:
is it possible to do this once in `settings.gradle.kts` and have it available everywhere? In which case I'd probably just specify the values themselves in there instead of being in the `gradle.properties` (with the exception of `toolkitVersion` that needs to be able to be written by CI.

Reply:
Nope, we don't have access to the get methods of any project or root project. We can put them in buildSrc, but would prefer to do that in a different PR

Reply:
makes sense - and the `buildSrc` option is probably a decent compromise.